docs/DeveloperGuide.md

```diff
@@ -1,34 +1,522 @@
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+The following are remaining steps to be taken to finish the set up:
 
-## Non-Functional Requirements
+1. Make sure that the version is configured as __JDK 11__.
+2. When prompted, __import__ the project as a __Gradle project__ (could take minutes to complete).
+3. Enter commands to ensure that PlanNUS functions as expected. You may refer to the _User Guide_ for valid commands.
 
-{Give non-functional requirements}
 
-## Glossary
 
-* *glossary item* - Definition
+## Design
 
-## Instructions for manual testing
+### Architecture
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/Architecture.png" alt="Architecture diagram of PlanNUS"/>
+</div>
+
+The ***Architecture Diagram*** given above explains the high-level design of PlanNUS. Below is a quick overview of each component.
+
+
+
+### Overview
+
+#### PlanNus
+
+`PlanNus` class contains the `main` and `run` method, which is responsible for
+
+  * At launch
+
+       * Loading all modules for AY2020/21 into PlanNUS
+
+       * Loading previous save file into PlanNUS if available
+
+       * Creation of entry point to available apps in PlanNUS
+
+  * While running
+
+       * Continuously prompt user for app selection
+
+  * At shut down
+
+       * Saving of user data into save file
+       * Clean up methods where necessary
+
+
+
+#### Global, Ui, Parser, Storage, Apps
+
+* The `global` package contains classes, exceptions and objects that are required across the whole app. 
+* The `ui` package contains the class that is responsible for sharing one `scanner` class across the whole app to prevent multiple IO streams
+* The `parser` package contains the class that handles user's app selection
+* The `storage` package handles loading and saving of user's data to a save file.
+* Packages for Available apps such as Academic Planner and CAP Calculator are stored within `apps` package
+
+
+
+### Project Structure
+
+Each package in the PlanNUS as given above follows the following file structure where applicable:
+
+* A functional class that acts as the entry point to that module
+* A parser class that parses user input into executable commands by PlanNUS
+
+* `commands`: A package that handles all executable commands given by parser
+* `commons`: A package with the utilities and shared classes across the parent package
+* `exceptions`: A package to handle all exceptions thrown across the parent package
```

Comment:
Maybe you can explain more about "App Entry Point" because it shows in your diagram but there is no explanation about this.